### PR TITLE
Sphinx foss2016a

### PR DESCRIPTION
**Pull request by RvDijk**

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 4 (4 easyconfigs in this PR)
node2408.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/a94cbd9c8d51cb44d78f7b7d7c5e7b17 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 3 out of 5 (4 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/27ec4e1878be870f68b991c3a09ea89b for a full test report.

**Comment by boegel** [comment]
failing tests reek of not taking into account that `$TMPDIR` may point to a symlink (similar to problem with IPython 5.0.0, cfr. https://github.com/hpcugent/easybuild-easyconfigs/pull/3368)

**Comment by hajgato** [comment]
I tried to build,  and I got exactly the same 4 tests to fail.

**Comment by boegel** [comment]
@hajgato was `$TMPDIR` a symlink to another location in your case?

**Comment by hajgato** [comment]
@boegel: `$TMPDIR` points to a local disk `/tmp/$USERNAME`` (on `/dev/sda3`)
`Sphinx` uses a `path` class, and most of the cases path is `os.path.abspath()`. So this might have to be changed to `os.path.realpath()`

**Comment by hajgato** [comment]
@boegel I did not look deep into the problem, but there is some error with `latex` and/or `pdflatex` commnads. If your system does not have those installed, the tests that fail here would be skipped. I suspect that the old EBs would also fail on those systems that have latex installed. (Just tried the newest `Sphinx 1.4.7`, exactly those 4 tests fails than for `1.4.6`.)

**Comment by boegel** [comment]
@hajgato so, maybe we should just disable those tests then? It's not the first time the Sphinx test suite causes problems, see https://github.com/hpcugent/easybuild-easyconfigs/issues/644...

**Comment by hajgato** [comment]
@boegel if we specifically switch off those tests (`latex` and `pdflatex`) then probably yes. I do not know how configurable the tests. Anyway, it is only a document generator, and who builds documents in HPC environment?

**Comment by boegel** [comment]
@hajgato Sphinx is sometimes required as a dependency for other stuff, for example to 'build' the included documentation...

**Comment by hajgato** [comment]
@boegel I just wanted to say that its not the endo fo the world, if you do not have a pdf documentation built from tex.

**Comment by RvDijk** [comment]
@boegel so should the tests of latex and pdflatex be turned off?

**Comment by boegel** [comment]
@RvDijk I think so, yes, they've been causing too much trouble... Not sure how easy that is though, are you up for figuring that out?

**Comment by RvDijk** [comment]
@boegel sure, I'll look into it.

File: easybuild/easyconfigs/s/Sphinx/Sphinx-1.4.6-foss-2016a-Python-2.7.11.eb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+exts_defaultclass = 'PythonPackage'
+
+dependencies = [
+    ('Python', '2.7.11'),
+    ('Pygments', '2.1.3', '-Python-%(pyver)s'),
+    ('requests', '2.10.0', versionsuffix),
+]
+
+patches = [
+    '%(name)s-%(version)s-pdflatex_failing_tests.patch'
```
Comment (boegel) [inline easybuild/easyconfigs/s/Sphinx/Sphinx-1.4.6-foss-2016a-Python-2.7.11.eb]: @RvDijk the patch will not be applied this way, you need to specify it in the `exst_list` entry for Sphinx itself below:

```python
(name, version, {
    'source_urls': ['https://pypi.python.org/packages/source/S/Sphinx/'],
    'patches': ['Sphinx-%(version)s-pdflatex_failing_tests.patch'],
}),
```

(I'm not sure `%(version)s` will work there, you'll need to check that)

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 4 (4 easyconfigs in this PR)
node2413.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.12
See https://gist.github.com/152ba13e9e57882aa67224bbf0b92d75 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2098.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/0072bdec1e44910d862678094f84c867 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2413.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/0b5b094a8978388bb4a73e9bd9e7d96c for a full test report.

**Comment by boegel** [comment]
lgtm, thanks for following up on this @RvDijk!

**Pull request merged**